File: 409-longest-palindrome/longestPalindrome.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

/**
 理解错了题意。。WA了两次。。
 */
int longestPalindrome(string s) {
    int ret = 0, mid = 0;
    unordered_map<char, int> myMap;
    for (char c : s) myMap[c]++;
    for (auto itor : myMap) {
        int num = itor.second;
        ret += num;
        if (num % 2 == 1) {
            ret--;
            mid = true;
        }
    }
    return mid ? ret + 1 : ret;
}

int main() {
    cout << longestPalindrome("aaa") << endl;
    return 0;
}

File: 409-longest-palindrome/longestPalindrome.py
from collections import defaultdict

class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        ret = 0
        mid = 0
        myDict = defaultdict(lambda: 0)
        for c in s: myDict[c] += 1
        for key, value in myDict.items():
            ret += value;
            if value % 2 == 1:
                ret -= 1
                mid = True;
        return ret + 1 if mid else ret

s = Solution()
print(s.longestPalindrome("aaaav"))
